cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,247 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	calculationsv1 "github.com/vega-project/ccb-operator/pkg/apis/calculations/v1"
+	"github.com/vega-project/ccb-operator/pkg/client/clientset/versioned/fake"
+	v1 "github.com/vega-project/ccb-operator/pkg/client/clientset/versioned/typed/calculations/v1"
+	"github.com/vega-project/ccb-operator/pkg/util"
+)
+
+func TestCreateCalculation(t *testing.T) {
+	fakecs := fake.NewSimpleClientset()
+	var fakeClient v1.VegaV1Interface
+	fakeClient = fakecs.VegaV1()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	s := server{
+		logger: logrus.WithField("test-name", "get calculations test"),
+		ctx:    ctx,
+		client: fakeClient,
+	}
+
+	var jsonStr = []byte(`{"teff": "10000.0", "logG": "4"}`)
+	req, err := http.NewRequest("POST", "/calculations/create", bytes.NewBuffer(jsonStr))
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(s.createCalculation)
+	handler.ServeHTTP(rr, req)
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v",
+			status, http.StatusOK)
+	}
+
+	type jsonSpec struct {
+		Spec map[string]interface{}
+	}
+
+	type testData struct {
```

Comment:
we don't need that. We need to compare the whole calculation object, not only teff and logG